etl/steps/data/garden/sdg/latest/sdg.py

```diff
@@ -60,13 +60,21 @@ def run(dest_dir: str) -> None:
     for dataset_name, sdg_group in sdg_sources.groupby("dataset_name"):
         ds = Dataset(DATA_DIR / "backport/owid/latest" / dataset_name)
 
+        # Since ds[table] reads from a feather file, it becomes the bottleneck in
+        # runtime. Caching saves us from repeated reads
+        table_cache: dict[str, Table] = {}
+
         # go over all indicators from that dataset
         for r in sdg_group.itertuples():
 
             # iterate over all tables in a dataset (backported datasets would
             # usually have only one)
             for table_name in ds.table_names:
-                table = ds[table_name]
+                if table_name in table_cache:
+                    table = table_cache[table_name]
```

Comment:
You can have a function like this in `sdg.py`

```python
import functools

@functools.lru_cache()
def _load_table_from_dataset(dataset_name: str, table_name: str) -> Table:
    """Load table from dataset."""
    ds = Dataset(DATA_DIR / "backport/owid/latest" / dataset_name)
    return ds[table_name]

...
table = _load_table_from_dataset(dataset_name, table_name)
```

When testing this, I realized that even better solution might be to switch for loops from
```
for r in sdg_group.itertuples():
    for table_name in ds.table_names:
        table = ds[table_name]
```

to

```
for table_name in ds.table_names:
    table = ds[table_name]
    for r in sdg_group.itertuples():
        ...
```

I haven't tested it though, so there might be a catch (and it might be slower than the caching solution)

Reply:
Anyway, your solution looks simpler than `functools.lru_cache`